lib/karto_sdk/src/Mapper.cpp

```diff
@@ -2548,22 +2601,49 @@ void Mapper::setParamUseResponseExpansion(bool b)
 
 void Mapper::Initialize(kt_double rangeThreshold)
 {
-  if (m_Initialized == false) {
-    // create sequential scan and loop matcher
-    m_pSequentialScanMatcher = ScanMatcher::Create(this,
-        m_pCorrelationSearchSpaceDimension->GetValue(),
-        m_pCorrelationSearchSpaceResolution->GetValue(),
-        m_pCorrelationSearchSpaceSmearDeviation->GetValue(),
-        rangeThreshold);
-    assert(m_pSequentialScanMatcher);
+  if (m_Initialized) {
+    return;
+  }
+
+  if (m_Deserialized) {
+    InitializeFromFile(rangeThreshold);
+    return;
+  }
+
+  // create sequential scan and loop matcher
+  m_pSequentialScanMatcher = ScanMatcher::Create(this,
+      m_pCorrelationSearchSpaceDimension->GetValue(),
+      m_pCorrelationSearchSpaceResolution->GetValue(),
+      m_pCorrelationSearchSpaceSmearDeviation->GetValue(),
+      rangeThreshold);
+  assert(m_pSequentialScanMatcher);
 
-    m_pMapperSensorManager = new MapperSensorManager(m_pScanBufferSize->GetValue(),
-        m_pScanBufferMaximumScanDistance->GetValue());
+  m_pMapperSensorManager = new MapperSensorManager(m_pScanBufferSize->GetValue(),
+      m_pScanBufferMaximumScanDistance->GetValue());
 
-    m_pGraph = new MapperGraph(this, rangeThreshold);
+  m_pGraph = new MapperGraph(this, rangeThreshold);
+
+  m_Initialized = true;
+}
 
-    m_Initialized = true;
+void Mapper::InitializeFromFile(kt_double rangeThreshold){
+  // create new sequential and loop scan matcher based on current config
+  if (m_pSequentialScanMatcher) {
+    delete m_pSequentialScanMatcher;
   }
+  m_pSequentialScanMatcher = ScanMatcher::Create(this,
```

Comment:
This `m_pSequentialScanMatcher` call is the same as in `Initialize`, so if you just moved the `if (m_Deserialized) {` down below it, you can use the creation of the new matcher and then this function can only do the specific `Set...()` and `Update...()` calls. 

You can also move the `if (m_pSequentialScanMatcher) {` to the initialize function as well I think?

Reply:
Thanks for pointing that out.
Since there was little left in `initializeFromFile()` after that, I decided to move those calls as well to have everything in one place.